packages/astro/src/assets/types.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { VALID_INPUT_FORMATS, VALID_OUTPUT_FORMATS } from './consts.js';
+import { ImageService } from './services/service.js';
+
+export type ImageQualityPreset = 'low' | 'mid' | 'high' | 'max' | (string & {});
+export type ImageQuality = ImageQualityPreset | number;
+export type InputFormat = (typeof VALID_INPUT_FORMATS)[number] | (string & {});
+export type OutputFormat = (typeof VALID_OUTPUT_FORMATS)[number] | (string & {});
+
+declare global {
+	// eslint-disable-next-line no-var
+	var astroAsset: {
+		imageService?: ImageService;
+		addStaticImage?: ((options: ImageTransform) => string) | undefined;
+		staticImages?: Map<ImageTransform, string>;
+	};
+}
+
+/**
+ * Type returned by ESM imports of images and direct calls to imageMetadata
+ */
+export interface ImageMetadata {
+	src: string;
+	width: number;
+	height: number;
+	format: InputFormat;
+}
+
+/**
+ * All the transformations possible to an image
```

Comment:
Is this the most accurate description of this type or would “transformations” better be framed as “paramaters” given that `src` isn’t really a transformation?

Something like this maybe?

```suggestion
 * Options accepted by the image transformation service.
```

Reply:
I like it!